src/DurableTask.Core/TaskOrchestrationContext.cs

```diff
@@ -571,12 +571,16 @@ public void HandleEventWhileSuspended(HistoryEvent historyEvent)
         public void HandleExecutionSuspendedEvent(ExecutionSuspendedEvent suspendedEvent)
         {
             this.IsSuspended = true;
-            
+
+            //When the orchestrator is suspended, a task could potentially be added to the orchestratorActionsMap.
+            //This could lead to the task being executed repeatedly without completion until the orchestrator is resumed.
+            //To prevent this scenario, check if orchestratorActionsMap is empty before proceeding.
```

Comment:
stylistic nit: let's always add a space between "//" and the text :) . This can be automated with an IDE style linter, would love to invest on that eventually.

```suggestion

            // When the orchestrator is suspended, a task could potentially be added to the orchestratorActionsMap.
            // This could lead to the task being executed repeatedly without completion until the orchestrator is resumed.
            // To prevent this scenario, check if orchestratorActionsMap is empty before proceeding.
```